prusti-viper/src/encoder/mir/specifications/specs.rs

```diff
@@ -145,13 +145,9 @@ impl<'tcx> Specifications<'tcx> {
             .cloned()
             .unwrap_or_else(ProcedureSpecification::empty);
 
-        let trait_spec = self.get_proc_spec(env, trait_query);
-
-        let refined = if let Some(trait_spec_set) = trait_spec {
-            impl_spec.refine(trait_spec_set)
-        } else {
-            impl_spec
-        };
+        let empty = ProcedureSpecification::empty();
+        let trait_spec = self.get_proc_spec(env, trait_query).unwrap_or(&empty);
+        let refined = impl_spec.refine(trait_spec);
```

Comment:
```suggestion
        let trait_spec = self.get_proc_spec(env, trait_query)
            .unwrap_or_else(ProcedureSpecification::empty);
        let refined = impl_spec.refine(trait_spec);
```

(Maybe even `unwrap_or_default` would work here?)

Reply:
Currently, `trait_spec` is required to be a reference due to the signature of `Refine::refine`, I could change the signature to take ownership instead which would allow me to simplify things.  I don't think this would have much impact on performance since in almost all cases the individual parts a cloned anyways (or it panics)

Reply:
Ah, never mind then, I missed the `&`.